Source/Modules/c.cxx

```diff
@@ -0,0 +1,3149 @@
+
+  virtual int enumvalueDeclaration(Node *n) {
+    if (Cmp(Getattr(n, "ismember"), "1") == 0 && Cmp(Getattr(n, "access"), "public") != 0)
+      return SWIG_NOWRAP;
+    Swig_require("enumvalueDeclaration", n, "?enumvalueex", "?enumvalue", NIL);
+
+    if (!GetFlag(n, "firstenumitem")) {
+      Printv(enum_decl, ",\n", NIL);
+      if (cxx_enum_decl)
+	Printv(cxx_enum_decl, ",\n", NIL);
+    }
+
+    String* const symname = Getattr(n, "sym:name");
+    Printv(enum_decl, cindent, enum_prefix_.get(), symname, NIL);
+    if (cxx_enum_decl)
+      Printv(cxx_enum_decl, cxx_enum_indent, cindent, cxx_enum_prefix_.get(), symname, NIL);
+
+    // We only use "enumvalue", which comes from the input, and not "enumvalueex" synthesized by SWIG itself because C should use the correct value for the enum
+    // items without an explicit one anyhow (and "enumvalueex" can't be always used as is in C code for enum elements inside a class or even a namespace).
+    String *value = Getattr(n, "enumvalue");
+    if (value) {
+      // We can't always use the raw value, check its type to see if we need to transform it.
+      maybe_owned_dohptr cvalue;
+      switch (SwigType_type(Getattr(n, "type"))) {
+	case T_BOOL:
+	  // Boolean constants can't appear in C code, so replace them with their values in the simplest possible case. This is not exhaustive, of course,
+	  // but better than nothing and doing the right thing is not simple at all as we'd need to really parse the expression, just textual substitution wouldn't
+	  // be enough (consider e.g. an enum element called "very_true" and another one using it as its value).
+	  if (Cmp(value, "true") == 0) {
+	    cvalue.assign_owned(NewString("1"));
+	  } else if (Cmp(value, "false") == 0) {
+	    cvalue.assign_owned(NewString("0"));
+	  } else {
+	    Swig_error(Getfile(n), Getline(n), "Unsupported boolean enum value \"%s\".\n", value);
+	  }
+	  break;
+
+	case T_CHAR:
+	  // SWIG parser doesn't put single quotes around char values, for some reason, so add them here.
+	  cvalue.assign_owned(NewStringf("'%(escape)s'", value));
+	  break;
+
+	default:
+	  cvalue.assign_non_owned(value);
+      }
+
+      Printv(enum_decl, " = ", cvalue.get(), NIL);
+      if (cxx_enum_decl)
+	Printv(cxx_enum_decl, " = ", cvalue.get(), NIL);
+    }
+
+    Swig_restore(n);
+    return SWIG_OK;
+  }
+
+  /* ---------------------------------------------------------------------
+   * constantWrapper()
+   * --------------------------------------------------------------------- */
+
+  virtual int constantWrapper(Node *n) {
+    String *name = Getattr(n, "sym:name");
+    // If it's a #define or a %constant, use raw value and hope that it will work in C as well as in C++. This is not ideal, but using "value" is even worse, as
+    // it doesn't even work for simple char constants such as "#define MY_X 'x'", that would end up unquoted in the generated code.
+    String *value = Getattr(n, "rawval");
```

Comment:
Once #2990 is merged, this will need updating.

The `rawval` attribute is gone, and `value` will be a valid C/C++ expression (so for a string or char literal it's suitably escaped and includes quotes).  For cases where we want to emit a string or character literal in the target language there's now `stringval` with the byte sequence for the string's value, but I think for C just using `value` should work in all cases we can actually reasonably handle.

I think consulting `staticmembervariableHandler:value` as below shouldn't then be necessary as there shouldn't be a situation where the `value` attribute isn't set but `staticmembervariableHandler:value` is.

So probably the line above (and the comment above it) and both the `if`s below can be replaced with just:

```
    String *value = Getattr(n, "value");
```

Reply:
Done, along with another fix to stop escaping `T_CHAR` values.

Reply:
Thanks Olly! I did plan to do it when merging the latest master into our branch but hadn't managed to do it before you did, thanks again!